### ISSUE TITLE
3 Notifications for 1 posting

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#9066]
### Expected behavior

Get only one notification.

### Actual behavior

Following situation with the current development code as of 96ffe95949f97a28354ff8f442e38b429e8c50a0

* I've set "get notifications on new postings" for the forum contact of the !Friendica Developers
* A new posting is made to that forum that contains an @-mention for me

and the result are three notifications shown in the notification drop down. Twice for the forum that made a new posting and once for the original author of that posting. But not that I was mentioned.

**Comment by tobiasd** [comment]
When a posting to the forum is done without mentioning me, then "only" two notifications are generated.

Could it be, that this is a new side-effect of the "detailed notification" setting?

**Comment by annando** [comment]
This is possible.

**Comment by tobiasd** [comment]
Indeed, turning off the "detailed notifications" collapses the notification thread into the "Forum has posted something". Before the recent changes, this was also the situation with the "detailed notifications" on, and since I only added the notification flag to the forum, but not the user who posted, I think it should be that way again. Where the notification about the mention is an actual benefit in my opinion.

**Comment by annando** [comment]
That's a side effect of the new new way the forums are working. Currently they are distributing the forum message via DFRN (for compatibility reasons), followed by an "Announce" (reshare) command via AP. I have to check how to work with this.

**Comment by tobiasd** [comment]
I just got 12 notifications for [this](https://libranet.de/display/0b6b25a8-115f-4a63-0521-266279617236) posting to the support forum. 1 for the creator, and 11 for the forum.

**Comment by annando** [comment]
It's complicate. I'm still thinking of the best way to handle it.

**Comment by tobiasd** [comment]
Since this mornings update to 3e95c7ef18423dd055604b234327d91af28e5de1 this problem seems to be solved. I'll keep an eye on it over the day and close it later, should it not reappear.

**Comment by annando** [comment]
This is more a coincidence. We have to do something at the notifications to really solve it.

**Comment by MrPetovan** [comment]
Is it possible Tobias got a notification for each appended `/Announce` item?

**Comment by annando** [comment]
Yes.

**Comment by MrPetovan** [comment]
Tentatively closed by #9098

**Comment by annando** [comment]
We have to have handled it when (hopefully at the end of 2020) will activate AP communication by the forums.

**Comment by tobiasd** [comment]
It happened again ;) also it was only two notification, one for the user who mentioned me and one for the forum as I requested that for new postings.

**Comment by annando** [comment]
Well, when it was a personal mention and the share notice, then it would be like expected, wouldn't it?

**Comment by MrPetovan** [comment]
With the detailed notifications, yes. If notifications are expected to be collapsed, no.

**Comment by tobiasd** [comment]
Only if one of the notifications would say "XYZ mentioned you in their posting" and the other "$ForumAccount posted something new" but they both say "ABC posted something new".

**Comment by tobiasd** [comment]
Also, a reply to a thread with a mention should not trigger the message "ABC replied to your posting" twice but either once, or correctly "replied to" and "mentioned you".

OTOH the detailed notification was for users who did not want to have their notifications collapsed but see one notification per different reply to the thread by different participants in the conversation. So I would much prefer a notification "mentioned you in a reply to your posting" then seeing two notifications for the same reply.
